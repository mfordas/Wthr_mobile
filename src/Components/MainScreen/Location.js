import React from 'react';
import { View, Text, TextInput, TouchableOpacity, Image, PermissionsAndroid } from 'react-native';
import { connect } from 'react-redux';

import { getCityCoordinatesByName, getCityNameByCoordinates } from '../../actions/locationActions';
import mainStyling from '../../main_styling/main_styling';

import locationSrc from '../../img/pin.png';
import confirmSrc from '../../img/checked.png';

class Location extends React.Component {
  constructor(props){
    super(props)

    this.state = {
      citySearch:'',
      showForm: false
    }
  }

  requestLocationPermission = async () => {
    try {
      const granted = await PermissionsAndroid.request(
        PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION,
        {
          title: "Wthr App Location Permission",
          message:
            "Wthr App needs access to your location ",
          buttonNeutral: "Ask Me Later",
          buttonNegative: "Cancel",
          buttonPositive: "OK"
        }
      );
      
      if (granted === PermissionsAndroid.RESULTS.GRANTED) {
        console.log("You can use the location");
      } else {
        console.log("Location permission denied");
      }
    } catch (err) {
      console.log('errror');
      console.warn(err);
    }
  };

  showCityName = () => {
    return this.props.cityData.cityData.city ? this.props.cityData.cityData.city : 'City not found'
  };

  showCityCountry = () => {
    return this.props.cityData.cityData.prov ? this.props.cityData.cityData.prov : 'try again'
  };

async componentDidMount () {
    this.requestLocationPermission();
};

  render() {
    return (
              <View><TouchableOpacity style={mainStyling.locationContainer} onPress={() => this.setState({showForm: !this.state.showForm})}>
          <Image style={mainStyling.iconLocation} source={locationSrc} />
          <Text style={mainStyling.mainText}>{this.showCityName()}, </Text>
          <Text style={mainStyling.mainText}>{this.showCityCountry()}</Text>
          </TouchableOpacity>
        { this.state.showForm ? <View style={mainStyling.inputContainer}>
          <TextInput style={mainStyling.input} onChangeText={text => this.setState({citySearch: text})}></TextInput>
          <TouchableOpacity style={mainStyling.locationContainer} onPress={async () => {
              this.setState({showForm: false});
              await this.props.getCityCoordinatesByName(this.state.citySearch);
              await this.props.getCityNameByCoordinates(this.props.coordinatesData.coordinatesData.latt, this.props.coordinatesData.coordinatesData.longt);
              }}>
          <Image style={mainStyling.iconConfirm} source={confirmSrc} />
          </TouchableOpacity>
        </View>: <></>}
        </View> 
    );
  }
}

const mapStateToProps = state => ({
  coordinatesData: state.coordinatesData,
  cityData: state.cityData,
})

export default connect(mapStateToProps, { getCityCoordinatesByName, getCityNameByCoordinates })(Location);