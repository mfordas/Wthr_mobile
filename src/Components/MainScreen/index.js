import React from 'react';
import { View, Text, TextInput, TouchableOpacity, Alert, Linking, Image, PermissionsAndroid, Button, ScrollView } from 'react-native';
import AsyncStorage from '@react-native-community/async-storage';
import Geolocation from '@react-native-community/geolocation';
import axios from 'axios';
import { connect } from 'react-redux';

import { getWeatherData } from '../../actions/weatherActions';
import { getCityCoordinatesByName } from '../../actions/locationActions';
import mainStyling from '../../main_styling/main_styling';
// import ErrorMessage from '../ReusableComponents/ErrorMessage';

import sunSrc from '../../img/sun.png';
import weatherSrc from '../../img/weather.png';
import weather2Src from '../../img/weather_2.png';
import cloudSrc from '../../img/cloud.png';
import cloudySrc from '../../img/cloudy.png';
import rainSrc from '../../img/rain.png';
import stormSrc from '../../img/storm.png';
import snowflakeSrc from '../../img/snowflake.png';
import fogSrc from '../../img/fog.png';
import foggySrc from '../../img/foggy.png';
import atmospheric2Src from '../../img/atmospheric_2.png';
import atmosphericSrc from '../../img/atmospheric.png';
import moonSrc from '../../img/moon.png';
import nightSrc from '../../img/night.png';
import gaugeSrc from '../../img/gauge.png';
import humiditySrc from '../../img/humidity.png';
import windSrc from '../../img/wind.png';

import locationSrc from '../../img/pin.png';
import confirmSrc from '../../img/checked.png';

class MainScreen extends React.Component {
  // constructor(props){
  //   super(props)

  //   this.state = {
  //     weatherImg: '',
  //     citySearch:''
  //   }
  // }

  


  chooseIcon(iconCode) {
    switch (iconCode) {
      case '01d':
        return <Image style={mainStyling.mainWeatherIcon} source={sunSrc} />;
      case '01n':
        return <Image style={mainStyling.mainWeatherIcon} source={nightSrc} />;
      case '02d':
        return <Image style={mainStyling.mainWeatherIcon} source={weatherSrc} />
      case '02n':
        return <Image style={mainStyling.mainWeatherIcon} source={moonSrc} />
      case '03d':
      case '03n':
        return <Image style={mainStyling.mainWeatherIcon} source={cloudSrc} />
      case '04d':
      case '04n':
        return <Image style={mainStyling.mainWeatherIcon} source={cloudySrc} />
      case '09d':
      case '09n':
        return <Image style={mainStyling.mainWeatherIcon} source={rainSrc} />
      case '10d':
        return <Image style={mainStyling.mainWeatherIcon} source={weather2Src} />
      case '10n':
        return <Image style={mainStyling.mainWeatherIcon} source={atmospheric2Src} />
      case '11d':
        return <Image style={mainStyling.mainWeatherIcon} source={stormSrc} />
      case '11n':
        return <Image style={mainStyling.mainWeatherIcon} source={atmosphericSrc} />
      case '13d':
      case '13n':
        return <Image style={mainStyling.mainWeatherIcon} source={snowflakeSrc} />
      case '50d':
        return <Image style={mainStyling.mainWeatherIcon} source={fogSrc} />
      case '50n':
        return <Image style={mainStyling.mainWeatherIcon} source={foggySrc} />
      default:
        return null
    }
  };

  requestLocationPermission = async () => {
    try {
      const granted = await PermissionsAndroid.request(
        PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION,
        {
          title: "Wthr App Location Permission",
          message:
            "Wthr App needs access to your location ",
          buttonNeutral: "Ask Me Later",
          buttonNegative: "Cancel",
          buttonPositive: "OK"
        }
        
      );
      
      if (granted === PermissionsAndroid.RESULTS.GRANTED) {
        console.log("You can use the location");
      } else {
        console.log("Location permission denied");
      }
    } catch (err) {
      console.warn(err);
    }
  };

  async getCityNameByCoordinates() {

    const mainData = await axios.get(`https://geocode.xyz/${this.state.lat},${this.state.lon}?geoit=json`);
    const cityData = mainData.data;

    this.setState({city: cityData.city, country: cityData.prov});
    
  };

  
  
  

componentDidMount () {
    this.requestLocationPermission();
    // Geolocation.getCurrentPosition(info => {this.setState({lat: info.coords.latitude, lon: info.coords.longitude})});
    this.props.getWeatherData(this.props.coordinatesData.coordinatesData.lat, this.props.coordinatesData.coordinatesData.lon);
    // this.chooseIcon(this.props.weatherData.weatherData.weather.icon);
};

// componentDidUpdate (prevProps, prevState) {
//   if((this.state.lat !== prevState.lat) && (this.state.lon !== prevState.lon)){
//   this.getCityNameByCoordinates();
//   this.props.getWeatherData();
// };

// };


  render() {

   
  
    return (
        <ScrollView style={mainStyling.container} contentContainerStyle={{flexGrow: 1, justifyContent: 'space-between'}}>
              <View><TouchableOpacity style={mainStyling.locationContainer}>
          <Image style={mainStyling.iconLocation} source={locationSrc} />
          <Text style={mainStyling.mainText}>, </Text>
          <Text style={mainStyling.mainText}></Text>
          </TouchableOpacity>
        <View style={mainStyling.inputContainer}>
          <TextInput style={mainStyling.input} onChangeText={text => this.props.getCityCoordinatesByName(text)}></TextInput>
          <TouchableOpacity style={mainStyling.locationContainer} onPress={() => {this.props.getWeatherData(this.props.coordinatesData.coordinatesData.latt, this.props.coordinatesData.coordinatesData.longt)}}>
          <Image style={mainStyling.iconConfirm} source={confirmSrc} />
          </TouchableOpacity>
        </View>
        </View> 
        <View style={mainStyling.weatherContainer}>
          <Text style={mainStyling.temperature}>{Math.round(this.props.weatherData.weatherData.main.temp)} Â°C</Text>
          <View style={mainStyling.conditionsContainer}>
          <Image style={mainStyling.conditionsIcon} source={windSrc} />
          <Text style={mainStyling.conditionsText}>{this.props.weatherData.weatherData.wind.speed} m/s</Text>
          </View>
          <View style={mainStyling.conditionsContainer}>
          <Image style={mainStyling.conditionsIcon} source={humiditySrc} />
    <Text style={mainStyling.conditionsText}>{this.props.weatherData.weatherData.main.humidity}%</Text>
          </View>
          <View style={mainStyling.conditionsContainer}>
          <Image style={mainStyling.conditionsIcon} source={gaugeSrc} />
          <Text style={mainStyling.conditionsText}>{this.props.weatherData.weatherData.main.pressure} hPa</Text>
          </View>
          {this.chooseIcon(this.props.weatherData.weatherData.weather[0].icon)}
        </View>
        <View style={mainStyling.iconsAuthorContainer}>
            <Text>Icons made by </Text>
            <TouchableOpacity onPress={() => Linking.openURL("https://www.flaticon.com/authors/those-icons")}><Text>Those Icons</Text></TouchableOpacity>
             <Text> from </Text>
             <TouchableOpacity onPress={() => Linking.openURL("https://www.flaticon.com/")}><Text>www.flaticon.com</Text></TouchableOpacity>
        </View>
        </ScrollView>
    );
  }
}

const mapStateToProps = state => ({
  weatherData: state.weatherData,
  coordinatesData: state.coordinatesData,
})

export default connect(mapStateToProps, { getWeatherData, getCityCoordinatesByName })(MainScreen);