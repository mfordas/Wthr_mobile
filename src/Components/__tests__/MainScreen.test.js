import 'react-native';
import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';

import MainScreen from '../MainScreen';
import mainStyling from '../../main_styling/main_styling';
import store from '../../store/store';

test('renders correctly', () => {
    const tree = renderer.create(<Provider store={store}><MainScreen /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });