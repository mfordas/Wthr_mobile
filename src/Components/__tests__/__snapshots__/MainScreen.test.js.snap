// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
      "justifyContent": "space-between",
    }
  }
  style={
    Object {
      "backgroundColor": "#03875b",
      "flex": 1,
    }
  }
>
  <View>
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "black",
            "flexDirection": "row",
            "fontSize": 15,
            "height": 50,
            "justifyContent": "center",
            "margin": 5,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "alignSelf": "center",
              "height": 20,
              "marginHorizontal": 10,
              "width": 20,
            }
          }
        />
        <Text
          style={
            Object {
              "alignSelf": "center",
              "fontFamily": "Exo2-VariableFont_wght",
              "fontSize": 20,
            }
          }
        >
          City not found
          , 
        </Text>
        <Text
          style={
            Object {
              "alignSelf": "center",
              "fontFamily": "Exo2-VariableFont_wght",
              "fontSize": 20,
            }
          }
        >
          try again
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "color": "black",
          "fontSize": 45,
          "justifyContent": "center",
          "margin": 0.5,
          "padding": 1,
          "textAlign": "center",
          "width": 200,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 70,
          }
        }
      >
        0
        Â°C
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 50,
            "marginHorizontal": 5,
          }
        }
      >
        <Image
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 40,
              "marginHorizontal": 15,
              "width": 40,
            }
          }
        />
        <Text
          style={
            Object {
              "fontFamily": "Exo2-VariableFont_wght",
              "fontSize": 20,
            }
          }
        >
          0
           
          km/h
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 50,
            "marginHorizontal": 5,
          }
        }
      >
        <Image
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 40,
              "marginHorizontal": 15,
              "width": 40,
            }
          }
        />
        <Text
          style={
            Object {
              "fontFamily": "Exo2-VariableFont_wght",
              "fontSize": 20,
            }
          }
        >
          0
          %
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 50,
            "marginHorizontal": 5,
          }
        }
      >
        <Image
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 40,
              "marginHorizontal": 15,
              "width": 40,
            }
          }
        />
        <Text
          style={
            Object {
              "fontFamily": "Exo2-VariableFont_wght",
              "fontSize": 20,
            }
          }
        >
          0
           hPa
        </Text>
      </View>
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 125,
            "margin": 15,
            "width": 125,
          }
        }
      />
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Go to 7 days forecast
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "color": "black",
          "flexDirection": "row",
          "fontSize": 10,
          "margin": 0.5,
          "padding": 1,
          "textAlign": "center",
        }
      }
    >
      <Text>
        Icons made by 
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text>
          Those Icons
        </Text>
      </View>
      <Text>
         from 
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text>
          www.flaticon.com
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
